schism/page_palette.c

```diff
@@ -256,6 +269,83 @@ static void palette_list_handle_key(struct key_event * k)
 		change_focus_to(n);
 }
 
+static void palette_copy_to_clipboard(void) {
+	char palette_text[49] = "";
+	palette_to_string(palette_text);
+	SDL_SetClipboardText((const char*)&palette_text);
+	status_text_flash("Palette copied to clipboard");
+}
+
+static void palette_paste_from_clipboard(void) {
+	if (SDL_HasClipboardText()) {
+		char* cb = SDL_GetClipboardText();
+
+		if(cb <= 0) {
+			printf("Failed getting clipboard text. Error: %s\n", SDL_GetError());
+			return;
+		}
+
+		printf("Pasted palette: \"%s\"\n", cb);
+
+		int i = 0;
+		int start = -1;
+		int end = -1;
+
+		// Remove spaces/newlines before and after content
+		while(1) {
```

Comment:
Edit `palette_trans` to have a null terminator,
```c
const char* palette = strpbrk(cb, palette_trans);
if (!palette) /* sanity check */
    return;
```
and then `set_palette_from_string()` should be able to handle it as is. You can also grab a version of `strpbrk` that takes in the length of the lookup array and shove it in util.c so you can use the translation table as-is.